### PR TITLE
[MNT] skpro as a soft dependency

### PR DESCRIPTION
**Pull request by fkiraly**
This PR makes skpro a soft dependency - this allows to develop and test integration with the reducers, e.g., reducers with a probabilistic prediction mode and an skpro estimator within, such as https://github.com/sktime/sktime/pull/5271

Long-term, we probably want skpro to be a core dependency and deduplicate the distribution modules.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -77,6 +77,7 @@ clustering = [
 forecasting = [
     "pmdarima>=1.8,!=1.8.1,<2.1",
     "prophet>=1.1,<1.2",
+    "skpro<2.1.0",
```
Comment (yarnabrina) [inline pyproject.toml:80]: Based on previous discussions, this must also have a lower bound.

File: pyproject.toml [hunk 2]
```diff
@@ -166,6 +167,7 @@ all_extras_pandas2 = [
     "scikit_posthocs>=0.6.5",
     "seaborn>=0.11.0",
     "seasonal",
+    "skpro<2.1.0",
```
Comment (yarnabrina) [inline pyproject.toml:170]: Same as above, but here quite a few doesn't have it so deprioritised (?)

**Review by yarnabrina — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (fkiraly) [reply pyproject.toml]: done

File: pyproject.toml (hunk 2 above)
Comment (fkiraly) [reply pyproject.toml]: done

**Comment by fkiraly** [comment]
comments addressed, kindly re-review

File: pyproject.toml [hunk 3]
```diff
@@ -77,6 +77,7 @@ clustering = [
 forecasting = [
     "pmdarima>=1.8,!=1.8.1,<2.1",
     "prophet>=1.1,<1.2",
+    "skpro<2.1.0,>=2.0.0",
```
Comment (yarnabrina) [inline pyproject.toml:80]:
```suggestion
    "skpro>=2.0.0,<2.1.0",
```

File: pyproject.toml [hunk 4]
```diff
@@ -166,6 +167,7 @@ all_extras_pandas2 = [
     "scikit_posthocs>=0.6.5",
     "seaborn>=0.11.0",
     "seasonal",
+    "skpro<2.1.0,>=2.0.0",
```
Comment (yarnabrina) [inline pyproject.toml:170]:
```suggestion
    "skpro>=2.0.0,<2.1.0",
```

**Review by yarnabrina — APPROVED** [review]
Added two completely optional non-blocking suggestions for consistency.

File: pyproject.toml (hunk 4 above)
Comment (fkiraly) [reply pyproject.toml:170]: why? also, why not...

**Review by yarnabrina — APPROVED** [review]

**Pull request merged**